OpenRA.Game/Graphics/SpriteRenderable.cs

```diff
@@ -94,21 +116,42 @@ public void Render(WorldRenderer wr)
 			if ((tintModifiers & TintModifiers.ReplaceColor) != 0)
 				a *= -1;
 
-			wsr.DrawSprite(sprite, palette, ScreenPosition(wr), scale, t, a);
+			wsr.DrawSprite(sprite, palette, ScreenPosition(wr), scale, t, a, rotation.RendererRadians());
 		}
 
 		public void RenderDebugGeometry(WorldRenderer wr)
 		{
 			var pos = ScreenPosition(wr) + sprite.Offset;
 			var tl = wr.Viewport.WorldToViewPx(pos);
 			var br = wr.Viewport.WorldToViewPx(pos + sprite.Size);
-			Game.Renderer.RgbaColorRenderer.DrawRect(tl, br, 1, Color.Red);
+			if (rotation == WAngle.Zero)
+				Game.Renderer.RgbaColorRenderer.DrawRect(tl, br, 1, Color.Red);
+			else
+				Game.Renderer.RgbaColorRenderer.DrawPolygon(Util.RotateQuad(tl, br - tl, rotation.RendererRadians()), 1, Color.Red);
 		}
 
 		public Rectangle ScreenBounds(WorldRenderer wr)
 		{
 			var screenOffset = ScreenPosition(wr) + sprite.Offset;
-			return new Rectangle((int)screenOffset.X, (int)screenOffset.Y, (int)sprite.Size.X, (int)sprite.Size.Y);
+			if (rotation == WAngle.Zero)
+				return new Rectangle((int)screenOffset.X, (int)screenOffset.Y, (int)sprite.Size.X, (int)sprite.Size.Y);
+			else
+			{
+				var rotatedQuad = Util.RotateQuad(screenOffset, sprite.Size, rotation.RendererRadians());
+				var minX = rotatedQuad[0].X;
+				var maxX = rotatedQuad[0].X;
+				var minY = rotatedQuad[0].Y;
+				var maxY = rotatedQuad[0].Y;
+				for (var i = 0; i < rotatedQuad.Length; i++)
+				{
+					minX = rotatedQuad[i].X < minX ? rotatedQuad[i].X : minX;
```

Comment:
At least in JS, If statement is more performant than Math.Min/Max. And given this is rendering code I think we should use the most performant function possible.
https://stackoverflow.com/questions/1232345/speed-and-style-of-math-max-vs-ternary-operator-in-javascript

Reply:
C# isn't JS!

I would hope that the JIT would be inlining `Math.Max` making the performance of the two options equivalent - so prefer readability unless you have profiling that shows otherwise!

Reply:
After researching a bit more it seems in C# it is either comparable or slightly better with `Math.Max`, so I will make the switch
https://stackoverflow.com/questions/5478877/math-max-vs-inline-if-what-are-the-differences

Reply:
This has been updated now.